src/apps/beefy/abis/BeefyClmVaults.sol

```diff
@@ -0,0 +1,44 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.2;
+
+interface ClmVault {
+    function balanceOf(address owner) external view returns (uint256);
+
+    function balances() external view returns (uint256, uint256);
+
+    function totalSupply() external view returns (uint256);
+
+    function wants() external view returns (address, address);
+}
+
+contract BeefyClmVaults {
+    struct VaultInfo {
+        address token0;
+        address token1;
+        uint256 amount0;
+        uint256 amount1;
+    }
+
+    function getUserVaults(
```

Comment:
I wrote this custom multicall contract. Technically, this could have been a normal multicall, the reason I did it this way is that I find it easier to read and work with on-chain data this way rather than having to manage the indexes for each call and then join it together again (ie for. each vault we have 4 calls, so we need to make sure to read the results in the right way).

The negative side of this approach is that we need to compile the contract to get the bytecode. I compiled it by writing the contract in another repo I have and then just copy-pasting the .sol file here, but maybe that's not the best approach. I can do it the other way if you think this complicates things too much.

Reply:
This seems ok, good to experiment!
Later we could automatically compile and update the bytecode to make it easier.

Reply:
Doesn't matter now, and please don't change, but if I remember correctly, one small advantage of the standard multicall with viem is that viem splits it into multiple RPC calls if the `data` part becomes too big.

https://viem.sh/docs/clients/public#batchmulticallbatchsize-optional

Reply:
That looks like a nice feature for multicalls :)